test/framework/easyblock.py

```diff
@@ -1512,6 +1512,73 @@ def test_patch_step(self):
         self.assertTrue("and very proud of it" in read_file(os.path.join(toydir, 'toy.source')))
         self.assertEqual(read_file(os.path.join(toydir, 'toy-extra.txt')), 'moar!\n')
 
+    def test_patch_step_dict(self):
+        """Test patch step."""
+        test_easyconfigs = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'easyconfigs', 'test_ecs')
+        ec = process_easyconfig(os.path.join(test_easyconfigs, 't', 'toy', 'toy-0.0-patch-step-dict.eb'))[0]
+
+        toy_patches = [
+            # test for dict syntax for patch list
+            {'filename': 'toy-0.0_fix-silly-typo-in-printf-statement.patch', 'opts': '--verbose'}
+        ]
+        self.assertEqual(ec['ec']['patches'], toy_patches)
+
+        # test applying patches without filename
+        ec['ec']['patches'] = [{'opts': '--verbose'}]
+        eb = EasyBlock(ec['ec'])
+        self.assertErrorRegex(EasyBuildError, 'filename is mandatory in dict-style patch', eb.fetch_step)
+
+        # test actual patching of unpacked sources
+        ec['ec']['patches'] = toy_patches
+        eb = EasyBlock(ec['ec'])
+        eb.fetch_step()
+        eb.extract_step()
+
+        # # patch step with captured output
+        # self.mock_stdout(True)
+        eb.patch_step()
+        # stdout = self.get_stdout()
+        # self.mock_stdout(False)
+        #
+        # # verify that patch cmd included opts ('--verbose')
+        # self.assertIn('opts: --verbose', stdout)
```

Comment:
How to check if additional options were included in the patch command?

Reply:
@Darkless012 The easiest way is probably by injecting a fake `patch` command that just writes its options to a file in `self.test_prefix`?

```python
fake_patch = os.path.join(self.test_prefix, 'patch')
fake_patch_out = os.path.join(self.test_prefix, 'patch.out')
write_file(fake_patch, "#!/bin/bash echo "$@" > %s" % fake_patch_out)
adjust_permissions(fake_patch, stat.S_IRUSR | stat.S_IXUSR)
os.environ['PATH'] = self.test_prefix + ':' + os.environ['PATH']

Reply:
Thanks, will test it out :)